client/core/core.go

```diff
@@ -1695,7 +1693,8 @@ func (c *Core) SetWalletPassword(appPW []byte, assetID uint32, newPW []byte) err
 		return newError(missingWalletErr, "wallet for %s (%d) is not known", unbip(assetID), assetID)
 	}
 
-	if newPasswordSet {
+	isSettingNewPW := newPW != nil // includes empty but non-nil
+	if isSettingNewPW {
```

Comment:
Makes me wonder what it means to do `core.SetWalletPassword(nil, ...)` and if that should be an error and documented as invalid.  What do you think @amassarwi?

Reply:
Agreed 💯

Reply:
will add a test for that as well

Reply:
done